tracing-appender/src/rolling.rs

```diff
@@ -0,0 +1,128 @@
+use crate::inner::InnerAppender;
+use chrono::{DateTime, Datelike, TimeZone, Timelike, Utc};
+use std::fmt::Debug;
+use std::io;
+use std::path::Path;
+
+#[derive(Debug)]
+pub struct RollingFileAppender {
+    inner: InnerAppender,
+}
+
+impl RollingFileAppender {
+    pub fn new(
+        rotation: Rotation,
+        directory: impl AsRef<Path>,
+        file_name_prefix: impl AsRef<Path>,
+    ) -> RollingFileAppender {
+        RollingFileAppender {
+            inner: InnerAppender::new(
+                directory.as_ref(),
+                file_name_prefix.as_ref(),
+                rotation,
+                Utc::now(),
+            )
+            .expect("Failed to create appender"),
+        }
+    }
+}
+
+impl io::Write for RollingFileAppender {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        self.inner.write(buf)
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        self.inner.flush()
+    }
+}
+
+/// Creates an hourly, rolling file appender.
+///
+/// The appender returned by `rolling::hourly` can be used with `non_blocking` to create
+/// a non-blocking, hourly file appender.
+/// # Examples
+/// ``` rust
+///
+/// fn main() {
+/// # fn doc() {
+/// let appender = tracing_appender::rolling::hourly("/some/path", "rolling.log");
+/// let (non_blocking_appender, _guard) = tracing_appender::non_blocking(appender);
+///
+/// let subscriber = tracing_subscriber::fmt().with_writer(non_blocking_appender);
+///
+/// tracing::subscriber::with_default(subscriber.finish(), || {
+///     tracing::event!(tracing::Level::INFO, "Hello");
+/// });
```

Comment:
since the `main` isn't hidden and will show up in the docs, let's indent this code properly under `main`:
```suggestion
///     let appender = tracing_appender::rolling::hourly("/some/path", "rolling.log");
///     let (non_blocking_appender, _guard) = tracing_appender::non_blocking(appender);
///
///     let subscriber = tracing_subscriber::fmt().with_writer(non_blocking_appender);
///
///     tracing::subscriber::with_default(subscriber.finish(), || {
///         tracing::event!(tracing::Level::INFO, "Hello");
///     });
```